src/adaptix/_internal/morphing/concrete_provider.py

```diff
@@ -75,6 +75,57 @@ def datetime_format_dumper(data: datetime):
         return datetime_format_dumper
 
 
+@dataclass
+@for_predicate(datetime)
+class DatetimeTimestampProvider(LoaderProvider, DumperProvider):
+    tz: typing.Optional[timezone] = None
+
+    def _provide_loader(self, mediator: Mediator, request: LoaderRequest) -> Loader:
+        tz = self.tz
+
+        def datetime_timestamp_loader(data):
+            try:
+                if data is None:
+                    raise TypeLoadError(float, data)
+
+                return datetime.fromtimestamp(data, tz=tz)
+            except TypeError:
+                raise TypeLoadError(float, data)
+            except OverflowError:
+                raise ValueLoadError(
+                    "Timestamp is out of the range of supported values",
+                    data,
+                )
+
+        return datetime_timestamp_loader
+
+    def _provide_dumper(self, mediator: Mediator, request: DumperRequest) -> Dumper:
+        def datetime_timestamp_dumper(data: datetime):
+            return data.timestamp()
+
+        return datetime_timestamp_dumper
+
+
+@for_predicate(date)
+class DateTimestampProvider(LoaderProvider):
+    def _provide_loader(self, mediator: Mediator, request: LoaderRequest) -> Loader:
+        def date_timestamp_loader(data):
+            try:
+                if data is None:
+                    raise TypeLoadError(float, data)
+
+                return date.fromtimestamp(data)  # noqa: DTZ012
+            except TypeError:
+                raise TypeLoadError(float, data)
+            except OverflowError:
+                raise ValueLoadError(
+                    "Timestamp is out of the range of supported values",
+                    data,
+                )
```

Comment:
Most of library users use interpreter version that does not need this extra checking.

So, we need to add extra validation if this snippet returns ``True``
```python
try:
    import _pydatetime
except ImportError:
     return False
else:
     return datetime is _pydatetime
```